base/src/main/scala/typeclass/InvariantFunctor.scala

```diff
@@ -0,0 +1,10 @@
+package scalaz
+package typeclass
+
+trait InvariantFunctor[F[_]] {
+  def xmap[A, B](ma: F[A])(f: A => B)(g: B => A): F[B]
```

Comment:
In general, `Iso` is a data type and `InvariantFunctor` is a type-class. I think we might (should?) agree to not conflate these two, going forward. That is, use a type-class excuslively so, and same for a data type.